packages/charts/src/scales/scale_band.test.ts

```diff
@@ -41,8 +41,8 @@ describe('Scale Band', () => {
     expect(scale.bandwidth).toBe(25);
     expect(scale.isValueInDomain('a')).toBe(true);
     expect(scale.isValueInDomain('b')).toBe(true);
-    expect(scale.isValueInDomain('z')).toBe(false);
-    expect(scale.isValueInDomain(null)).toBe(false);
+    expect(scale.isValueInDomain(('z' as unknown) as 'a' | 'b' | 'c' | 'd')).toBe(false);
+    expect(scale.isValueInDomain((null as unknown) as 'a' | 'b' | 'c' | 'd')).toBe(false);
```

Comment:
I think we could be looser on using `any` inside tests because unit tests are for the logic and not the types in my eyes. Especially in this case where you are casting something that is obviously not the right type for testing a js edge case.

In this case we could say `@ts-expect-error` with a description to show this was intentional and why.

```ts
    // @ts-expect-error: bad input
    expect(scale.isValueInDomain('z')).toBe(false);
    expect(scale.isValueInDomain(null)).toBe(false);
```

the issue with this is that it applied to the whole line so you don't get type checking outside of the one bad variable. If there were more than one intentional error I think this usage would be good but in this particular case I'm in favor of using `any`.

```suggestion
    expect(scale.isValueInDomain('z' as any)).toBe(false);
    expect(scale.isValueInDomain(null)).toBe(false);
```

Reply:
I agree, `any` is absolutely fine by me in tests, in fact usually better